lib/rules/flow-exact-props.js

```diff
@@ -0,0 +1,98 @@
+const t = require("@babel/types");
+
+const isReactClassComponent = node => {
+    if (t.isClassDeclaration(node)) {
+        const {superClass} = node;
+        if (t.isMemberExpression(superClass)) {
+            const {object, property} = superClass;
+            if (
+                t.isIdentifier(object, {name: "React"}) &&
+                t.isIdentifier(property, {name: "Component"})
+            ) {
+                return true;
+            }
+        }
+    }
+    return false;
+};
+
+const isReactFunctionalComponent = node => {
+    if (t.isArrowFunctionExpression(node)) {
+        if (
+            node.params.length === 1 &&
+            t.isIdentifier(node.params[0], {name: "props"})
+        ) {
+            return true;
+        }
+    }
+};
+
+const maybeReport = (node, props, typeAliases, context) => {
+    if (t.isGenericTypeAnnotation(props)) {
+        const {name} = props.id;
+        const alias = typeAliases.get(name);
+        if (alias && !alias.right.exact) {
+            const sourceCode = context.getSource();
+            context.report({
+                fix(fixer) {
+                    const right = alias.right;
+                    const rightText = sourceCode.slice(
+                        right.range[0] + 1,
+                        right.range[1] - 1,
+                    );
+                    const replacementText = `{|${rightText}|}`;
```

Comment:
So this is a case where it can auto-fix, but the fix may generate errors (from Flow). Just something for us to make sure we communicate clearly.

Reply:
I can make a note of this in the markdown file.